pkg/build/builder/docker.go

```diff
@@ -69,12 +70,22 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
+		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
+		d.build.Status.Message = api.StatusMessageFetchSourceFailed
+		if updateErr := updateStatusRetrier(d.build, d.client, d.build.Spec.Revision); updateErr != nil {
```

Comment:
yes, that was exactly what i was wondering but i wasn't sure that you literally want me to pass (nil) like that. i was afraid of the same thing, what if the build revision is already set previously somewhere else throughout the flow.
anyway. fixing now.